Allow the script to print the results on stdout as well

d00rman opened the pull request:
Apart from sending the e-mail, the script should also be able to print the results on stdout so that an operator can run the script on demand without too much e-mail traffic.

This PR adds two such modes: csv and pretty-printed.

d00rman committed:
Allow the script to print the results on stdout as well

Apart from sending the e-mail, the script should also be able to print
the results on stdout so that an operator can run the script on demand
without too much e-mail traffic.

This commit adds two such modes: csv and pretty-printed.

d00rman commented:
I plan to put this version in Puppet. @eevans could you review?

eevans commented on report-topk-partion-sizes:
```diff
@@ -351,13 +347,58 @@ def strfsize(num, suffix='B'):
-    parser.add_argument("-e", "--email", metavar="ADDRESS", required=True, help="Email address to send report to")
+    parser.add_argument("-e", "--email", metavar="ADDRESS", default='', help="Email the report to the specified address")
+    parser.add_argument('-s', '--csv', action='store_true', help='Output a comma-delimited report to stdout')
+    parser.add_argument('-t', '--printed', action='store_true', help='Output a pretty-formatted report to stdout')
```
A minor nit, but why not make the absence of `-e`/`-s` flags result in printing to standard out?

d00rman replied:
Euh, because the absence of one of them constitutes an error, and errors are to be output on stderr, aren't they?

eevans replied:
Not sure what you mean; My thinking was that printing a formatted report to stdout sounded like reasonable default behavior (i.e. in the absence of either email or csv).

d00rman replied:
Ah, ok! I misunderstood your question. Makes sense, will change.

eevans reviewed (changes requested):
One issue (other than the minor nit in-line): `logging` is using defaults; It'd be better if log output could be separated from the reports (csv in particular).  Something like `logging.basicConfig(stream=sys.stderr)` should do it.

d00rman commented:
> One issue (other than the minor nit in-line): logging is using defaults; It'd be better if log output could be separated from the reports (csv in particular). Something like logging.basicConfig(stream=sys.stderr) should do it.

Good idea, will fix!

d00rman committed:
Use Python's logger for informational and error messages

Note: the default level is set to WARNING because on INFO a lot of
annoying messages from urllib3 are being printed.

d00rman commented:
@eevans switched to using `logging.X(...)`

d00rman committed:
Output the report pretty-formatted to stdout by default

d00rman commented:
@eevans {{done}}

eevans reviewed (approved):
LGTM; Thanks!
